ledger/storage/transaction.go

```diff
@@ -40,9 +41,25 @@ func prefixkey(prefix byte, key []byte) []byte {
 	return k
 }
 
+func (m *TransactionManager) lockOnID(id *record.ID) {
+	m.db.idlocker.Lock(id)
+	m.locks = append(m.locks, id)
+}
+
+func (m *TransactionManager) releaseLocks() {
+	for _, id := range m.locks {
+		m.db.idlocker.Unlock(id)
+	}
+}
+
 // Commit tries to write transaction on disk. Returns error on fail.
 func (m *TransactionManager) Commit() error {
-	return m.txn.Commit(nil)
+	err := m.txn.Commit(nil)
+	m.releaseLocks()
```

Comment:
@andreyromancev probably locks should be released before Commit

Reply:
fixed in https://github.com/insolar/insolar/pull/363/commits/45bb695cdb50a03b3daa72bea8a90a9b1ff3353c